sklearn/preprocessing/data.py

```diff
@@ -1946,3 +1958,202 @@ def transform(self, X):
+                         "on positive data")
+    n_features = X.shape[1]
+    outputs = Parallel(n_jobs=-1)(delayed(_boxcox)(X, i, lambda_x=None)
+                                  for i in range(n_features))
+    output = np.concatenate([o[0][..., np.newaxis] for o in outputs], axis=1)
+    return output
+
+
+class BoxCoxTransformer(BaseEstimator, TransformerMixin):
+    """BoxCox transformation on individual features.
+
+    Boxcox transform wil be applied on each feature (each column of
+    the data matrix) with lambda evaluated to maximise the log-likelihood
+
+    Parameters
+    ----------
+    transformed_features : "all" or array of indices or mask
+        Specify what features are to be transformed.
+
+        - "all" (default): All features are to be transformed.
+        - array of int: Array of feature indices to be transformed..
+        - mask: Array of length n_features and with dtype=bool.
+
+    copy : boolean, optional, default True
+        Set to False to perform inplace computation.
+
+    Attributes
+    ----------
+    transformed_features_ : array of int
+        The indices of the features to be transformed
+    lambdas_ : array of float, shape (n_transformed_features,)
+        The parameters of the BoxCox transform for the selected features.
+    n_features_ : int
+        Number of features in input during fit
+
+    Notes
+    -----
+    The Box-Cox transform is given by::
+
+        y = (x ** lmbda - 1.) / lmbda,  for lmbda > 0
+            log(x),                     for lmbda = 0
+
+    ``boxcox`` requires the input data to be positive.
+
+    """
+    def __init__(self, transformed_features="all", n_jobs=1, copy=True):
+        self.transformed_features = transformed_features
+        self.n_jobs = n_jobs
+        self.copy = copy
+
+    def fit(self, X, y=None):
+        """Estimate lambda for each feature to maximise log-likelihood
+
+        Parameters
+        ----------
+        X : array-like, shape [n_samples, n_features]
+            The data to fit by apply boxcox transform,
+            to each of the features and learn the lambda.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        X = check_array(X, ensure_2d=True, dtype=FLOAT_DTYPES)
+        self.n_features_ = X.shape[1]
+        if self.transformed_features is "all":
+            self.transformed_features_ = np.arange(self.n_features_)
+        else:
+            self.transformed_features_ = np.copy(self.transformed_features)
+            if self.transformed_features_.dtype == np.bool:
+                self.transformed_features_ = \
+                    np.where(self.transformed_features_)[0]
+        if any(np.any(X[:, self.transformed_features_] <= 0, axis=0)):
+            raise ValueError("BoxCox transform can only be applied "
+                             "on positive data")
+        out = Parallel(n_jobs=self.n_jobs)(delayed(_boxcox)(X, i,
```

Comment:
Sorry, no haven't benchmarked this parallel method in either ``fit`` or ``transform``. Will try doing it and let you know the results.